ir/program/modifier.rs

```diff
@@ -35,22 +42,37 @@ impl Filter {
         }
         Ok(Self { variables: filter_variables })
     }
+
+    pub(crate) fn new_given_variable_map(
+        variables: Vec<&str>,
+        variable_index: &HashMap<String, Variable>,
+    ) -> Result<Self, ModifierDefinitionError> {
+        use ModifierDefinitionError::FilterVariableNotAvailable;
```

Comment:
Notice there's nothing linking `NotAvailable` with the `variable_index` and `given_variable_map()`... If i guessed i'd say this function was intended to create a filter for named variables given a map of available variables?

Looking at the two constructors, I see some similarities... can we merge them? (yes)

Then what's a better name for the function & the argument